l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -263,51 +284,79 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         returns (L2TransactionRequestTwoBridgesInner memory request)
     {
         require(l2BridgeAddress[_chainId] != address(0), "ShB l2 bridge not deployed");
-
-        (address _l1Token, uint256 _depositAmount, address _l2Receiver) = abi.decode(
-            _data,
-            (address, uint256, address)
-        );
-        require(_l1Token != L1_WETH_TOKEN, "ShB: WETH deposit not supported");
-        require(BRIDGE_HUB.baseToken(_chainId) != _l1Token, "ShB: baseToken deposit not supported");
-
-        uint256 amount;
-        if (_l1Token == ETH_TOKEN_ADDRESS) {
-            amount = msg.value;
-            require(_depositAmount == 0, "ShB wrong withdraw amount");
-        } else {
-            require(msg.value == 0, "ShB m.v > 0 for BH d.it 2");
-            amount = _depositAmount;
-
-            uint256 withdrawAmount = _depositFunds(_prevMsgSender, IERC20(_l1Token), _depositAmount);
-            require(withdrawAmount == _depositAmount, "5T"); // The token has non-standard transfer logic
+        bytes32 _assetInfo;
+        bytes memory _assetData;
+        try this.decodeLegacyData(_data, _prevMsgSender) returns (
+            bytes32 _assetInfoCatch,
+            bytes memory _assetDataCatch
+        ) {
+            (_assetInfo, _assetData) = (_assetInfoCatch, _assetDataCatch);
+        } catch {
+            (_assetInfo, _assetData) = abi.decode(_data, (bytes32, bytes));
         }
-        require(amount != 0, "6T"); // empty deposit amount
 
-        bytes32 txDataHash = keccak256(abi.encode(_prevMsgSender, _l1Token, amount));
-        if (!hyperbridgingEnabled[_chainId]) {
-            chainBalance[_chainId][_l1Token] += amount;
-        }
+        require(BRIDGE_HUB.baseToken(_chainId) != assetAddress[_assetInfo], "ShB: baseToken deposit not supported");
```

Comment:
This require seems meaningless since `assetAddress` is native token vault basically and for vast majority of tokens it will never hold. It is better to maintain a mapping of `uint256 chainId => bytes32 nativeAssetInfo`